src/Combinatorics/PhylogeneticTrees.jl

```diff
@@ -0,0 +1,289 @@
+mutable struct PhylogeneticTree{T <: Union{Float64, QQFieldElem}}
+  pm_ptree::Polymake.LibPolymake.BigObjectAllocated
+end
+
+function pm_object(PT::PhylogeneticTree)
+  return PT.pm_ptree
+end
+
+@doc raw"""
+    phylogenetic_tree(T::Type{<:Union{Float64, QQFieldElem}}, newick::String)
+
+Constructs a phylogenetic tree with Newick representation `newick`. `T` indicates 
+the numerical type of the edge lengths.
+
+# Examples
+Make a phylogenetic tree with 4 leaves from its Newick representation and print 
+its taxa and cophenetic matrix.
+```jldoctest
+julia> phylo_t = phylogenetic_tree(Float64, "((H:3,(C:1,B:1):2):1,G:4);");
+
+julia> taxa(phylo_t)
+4-element Vector{String}:
+ "B"
+ "C"
+ "G"
+ "H"
+
+julia> cophenetic_matrix(phylo_t)
+4×4 Matrix{Float64}:
+ 0.0  2.0  8.0  6.0
+ 2.0  0.0  8.0  6.0
+ 8.0  8.0  0.0  8.0
+ 6.0  6.0  8.0  0.0
+```
+"""
+function phylogenetic_tree(T::Type{<:Union{Float64, QQFieldElem}}, newick::String)
+  pm_ptree = Polymake.graph.PhylogeneticTree{Polymake.convert_to_pm_type(T)}(NEWICK = newick)
+
+  # load graph properties
+  pm_ptree.ADJACENCY
+  
+  return PhylogeneticTree{T}(pm_ptree)
+end
+
+@doc raw"""
+   phylogenetic_tree(M::Matrix{T}, taxa::Vector{String}) where T <: Union{Float64, QQFieldElem}
+
+Constructs a phylogenetic tree with cophenetic matrix `M` and taxa `taxa`. The matrix `M` must be
+ultrametric, otherwise an error will be thrown.
+
+# Examples
+Make a phylogenetic tree on 4 taxa with given cophenetic matrix and print one Newick representation.
+
+```jldoctest
+julia> mat = [0. 2 8 6; 2 0 8 6; 8 8 0 8; 6 6 8 0]
+4×4 Matrix{Float64}:
+ 0.0  2.0  8.0  6.0
+ 2.0  0.0  8.0  6.0
+ 8.0  8.0  0.0  8.0
+ 6.0  6.0  8.0  0.0
+
+julia> tax = ["Bonobo", "Chimpanzee", "Gorilla", "Human"]
+4-element Vector{String}:
+ "Bonobo"
+ "Chimpanzee"
+ "Gorilla"
+ "Human"
+
+julia> tree_mat = phylogenetic_tree(mat, tax);
+
+julia> newick(tree_mat)
+"Gorilla:4,(Human:3,(Bonobo:1,Chimpanzee:1):2):1;"
+```
+"""
+function phylogenetic_tree(M::Matrix{Float64}, taxa::Vector{String})
+  n_taxa = length(taxa)
+  @req (n_taxa, n_taxa) == size(M) "Number of taxa should match the rows and columns of the given matrix"
+  pm_ptree = Polymake.graph.PhylogeneticTree{Float64}(COPHENETIC_MATRIX = M, TAXA = taxa)
+  return PhylogeneticTree{Float64}(pm_ptree)
+end
+
+function phylogenetic_tree(M::QQMatrix, taxa::Vector{String})
+  n_taxa = length(taxa)
+  @req (n_taxa, n_taxa) == size(M) "Number of taxa should match the rows and columns of the given matrix"
+  pm_T = Polymake.convert_to_pm_type(QQFieldElem)
+  cp_M = convert(Matrix{pm_T}, Matrix{QQFieldElem}(M))
+
+  pm_ptree = Polymake.graph.PhylogeneticTree{pm_T}(
```

Comment:
```suggestion
  pm_ptree = Polymake.graph.PhylogeneticTree{Rational}(
```
We can pass a julia Rational here to get the corresponding polymake Rational type. (or even leave it out, since the default is Rational as well)